stdlib/arg.ml

```diff
@@ -132,7 +132,12 @@ let float_of_string_opt x =
 let parse_and_expand_argv_dynamic_aux allow_expand current argv speclist anonfun errmsg =
   let b = Buffer.create 200 in
   let initpos = !current in
-  let stop error =
+  let convert_error error =
+    (* convert an internal error to a Bad/Help exception
+       *or* add the program name as a prefix and the usage message as a suffix
+       to an user-raised Bad exception.
+    *)
+    let () = Buffer.clear b in
```

Comment:
I think that declaring `b` to be local to this function, instead of carefully clearing it, would make the code simpler and more robust.